src/string.cr

```diff
@@ -1742,6 +1742,18 @@ class String
     end
   end
 
+  # Returns a string where all chars in the given hash are replaced
+  # by the corresponding *tuple* values.
+  #
+  # ```
+  # "hello".gsub({'e' => 'a', 'l' => 'd'}) # => "haddo"
+  # ```
+  def gsub(tuple : NamedTuple)
+    gsub do |char|
+      tuple[char.to_s]? || char
```

Comment:
Yeah, let's do it in a separate commit and get this merged. Apologies for the delay @jhass